src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -97,3 +101,53 @@ class SubSystemLogicalTreeNode(var getOMInterruptDevice: (ResourceBindingsMap) =
     )
   }
 }
+
+class MemoryLogicalTreeNode[T <: Data](sram: DescribedSRAM[T]) extends LogicalTreeNode {
+
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
+    val omMemory = DiplomaticObjectModelAddressing.makeOMMemory(sram)
+
+    Seq(omMemory)
+  }
+}
+
+class ImpLogicalTreeNode extends LogicalTreeNode {
+  override def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent] = Nil): Seq[OMComponent] =
+    children
+}
+
+class TestMemoryLogicalTreeNode(device: () => Device, ramArchitecture: RAMArchitecture) extends LogicalTreeNode {
+  def getOMTestMemory(resourceBindings: ResourceBindings): Seq[OMComponent] = {
+    val memRegions : Seq[OMMemoryRegion]= DiplomaticObjectModelAddressing.getOMMemoryRegions("OMMemory", resourceBindings, None)
+    val Description(name, mapping) = device().describe(resourceBindings)
+    val ints = DiplomaticObjectModelAddressing.describeInterrupts(name, resourceBindings)
+
+    Seq[OMComponent](
+      OMTestMemory(
+        memoryRegions = memRegions,
+        interrupts = ints,
+        specifications = Nil,
+        memories = Nil,
+        architecture = ramArchitecture
+      )
+    )
+  }
+
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
+    val c = DiplomaticObjectModelAddressing.getOMComponentHelper(device(), resourceBindingsMap, getOMTestMemory)
+    val cc = components.map(_.asInstanceOf[OMMemory])
+    val children = c.toList.head.asInstanceOf[OMTestMemory].copy( memories = cc).asInstanceOf[OMComponent]
```

Comment:
This feels awkward, having to take the `c` object, which we just happen to know is a Seq containing exactly one `OMTestMemory` and having to grab the first element and cast it to an `OMTestMemory`. I know that the type system only knows that `c` is a Seq of `OMComponent`s, but this seems like we are throwing away type safety.

I think there's a way to make this work without explicit type casts by adding a type parameter to `getOMComponentHelper`, but I don't know if that refactoring is a good idea here.